src/ucp/core/ucp_worker.inl

```diff
@@ -56,6 +58,50 @@ ucp_worker_get_ep_by_id(ucp_worker_h worker, ucs_ptr_map_key_t id)
     return ep;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_ptr_map_key_t
+ucp_worker_get_request_id(ucp_worker_h worker, ucp_request_t *req, int indirect)
+{
+    ucs_ptr_map_key_t id;
+    ucs_status_t status;
+
+    status = ucs_ptr_map_put(&worker->ptr_map, req, indirect, &id);
+    if (ucs_unlikely(indirect)) {
+        return (status == UCS_OK) ? id : UCP_REQUEST_ID_INVALID;
+    }
+
+    ucs_assert(status == UCS_OK);
+    return id;
```

Comment:
why not just
return (status==OK) ? id : UCP_REQUEST_ID_INVALID

Reply:
because inline `ucs_ptr_map_put` already has `indirect` branch inside but 2 points where UCS_OK can be returned, so result code has to be simpler this way

Reply:
ok